### PR TITLE
insights: series based backfilling initial steps

### PR DESCRIPTION
**Pull request by chwarwick**
Setup the initial steps for a backfill pipeline that runs over a single repo.

The test is mostly to ensure the steps are connected in the pipeline and give a sample of how to use.  I anticipate that most of the test work will be done against each individual step.

## Test plan
n/a currently this code is not used
Unit test added to ensure pipeline steps are connected.

### CODE REVIEW

File: enterprise/internal/insights/pipeline/backfill.go [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+		output := make(chan SearchJobGeneratorOutput, 12)
+		buildJob := makeHistoricalSearchJobFunc(logger, commitClient)
+
+		var wg sync.WaitGroup
+		go func() {
+			wg.Wait()
+			close(output)
+		}()
+		// launching a single worker
+		wg.Add(1)
```
Comment (coury-clark) [inline enterprise/internal/insights/pipeline/backfill.go:98]: You might want to check out the package https://sourcegraph.sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/lib/group/group.go?L4%3A12=&utm_product_name=GoLand&utm_product_version=GoLand

File: enterprise/internal/insights/pipeline/backfill_test.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	}
+}
+
+func searchRunner(ctx context.Context, input <-chan SearchJobGeneratorOutput) <-chan SearchResultOutput {
+	output := make(chan SearchResultOutput)
+	goroutine.Go(func() {
+		defer close(output)
+		for job := range input {
+			output <- SearchResultOutput{
+				result:          searchResult{count: 10, capture: "", repo: job.Repo},
```
Comment (coury-clark) [inline enterprise/internal/insights/pipeline/backfill_test.go:41]: I think it would be useful to open issues to fill in all these gaps

**Review by coury-clark — APPROVED** [review]

File: enterprise/internal/insights/pipeline/backfill_test.go (hunk 2 above)
Comment (chwarwick) [reply enterprise/internal/insights/pipeline/backfill_test.go:41]: 👍  I have an issue open for this just moved it in progress.

File: enterprise/internal/insights/pipeline/backfill.go (hunk 1 above)
Comment (chwarwick) [reply enterprise/internal/insights/pipeline/backfill.go:98]: Nice, I remember seeing those before but forgot about them. I'll add a new issue for adding some parallelism in.

**Pull request merged**